src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,197 @@
+
+        final int areaToSearch = this.confirmArea(searchPatientDescriptor);
+        ArrayList<Person> personWithinTemperatureRange = new ArrayList<>();
+        List<Person> personList = model.getFilteredPersonList();
+
+        if (areaToSearch == areaNotFound) {
+            throw new CommandException(MESSAGE_NOT_FOUND);
+        }
+
+        else if (areaToSearch == areaIsName) {
+            String nameToSearch = searchPatientDescriptor.getName().toString().trim().toLowerCase();
+            Person personFound = personList.get(0);
+            boolean isPersonFound = false;
+
+            for (Person person : personList) {
+                String personName = person.getName().toString().trim().toLowerCase();
+                if (personName.equals(nameToSearch)) {
+                    personFound = person;
+                    isPersonFound = true;
+
+                }
+            }
+            if (!isPersonFound) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(String.format(MESSAGE_SEARCH_PERSON_SUCCESS, personFound));
+            }
+        }
+
+        else if (areaToSearch == areaIsTemperature) {
+            double startingTemperature = searchPatientDescriptor.getTemperatureRange().getStartingTemperature();
+            double endingTemperature = searchPatientDescriptor.getTemperatureRange().getEndingTemperature();
+            for (Person person : personList) {
+                if (person.getTemperature().getValue() >= startingTemperature
+                        && person.getTemperature().getValue() <= endingTemperature) {
+                    personWithinTemperatureRange.add(person);
+                }
+            }
+
+            if (personWithinTemperatureRange.isEmpty()) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(MESSAGE_SEARCH_PERSON_LIST_SUCCESS
+                        + getListOutput(personWithinTemperatureRange));
+            }
+        } else if (areaToSearch == tooManyArea) {
+            return new CommandResult(Messages.MESSAGE_TOO_MANY_COMMANDS);
+        }
+        return null;
+    }
+
+    /**
+     * Return the list of persons' details
+     * @param list a list that stores the filtered persons.
+     * @return a String output of the persons' details.
+     */
+    public String getListOutput(ArrayList<Person> list) {
+        StringBuilder outputString = new StringBuilder();
+        for (int i = 0; i < list.size(); i++) {
+            outputString.append(String.format("%d.%s\n", i + 1, list.get(i)));
+        }
+        return outputString.toString();
+    }
+
+    /**
+     * Confirm the area to look for.
+     * @param searchPatientDescriptor Details of the searchPatient Command.
+     * @return area to look for.
+     */
+    public int confirmArea(SearchPatientDescriptor searchPatientDescriptor) {
```

Comment:
Perhaps we can rename this method to `confirmSearchConditions` or `determineSearchCriteria`? `area` may not be very clear that we are referring to the search conditions.

Reply:
Hmm.. I feel both method name is ok